packages/core/src/components/sections/CrossSellingShelf/CrossSellingShelf.tsx

```diff
@@ -12,13 +14,25 @@ interface Props {
 }
 
 const CrossSellingShelf = ({ items, title, context, kind }: Props) => {
+  const { ref, inView } = useInView()
+
   const selectedFacets = useMemo(
     () => [{ key: kind, value: context.isVariantOf.productGroupID }],
     [kind, context.isVariantOf.productGroupID]
   )
 
   return (
-    <ProductShelf first={items} title={title} selectedFacets={selectedFacets} />
+    <Section
+      className={`${styles.section} section-product-shelf layout__section`}
+      ref={ref}
+    >
+      <UIProductShelf
```

Comment:
Components from `faststore/ui` we are using `UI` prefix. As this component comes from the core, we are using `ProductShelf`.

Reply:
Since we don't have a consensus about this, I think it better we let like that